components/forms/CreateTxForm/MsgForm/EncodeObjectForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { EncodeObject } from "@cosmjs/proto-signing";
+import { assert } from "@cosmjs/utils";
+import { useEffect, useState } from "react";
+import { MsgGetter } from "..";
+import { useAppContext } from "../../../../context/AppContext";
+import { isEncodeObject } from "../../../../lib/txMsgHelpers";
+import { TxMsg } from "../../../../types/txMsg";
+import Input from "../../../inputs/Input";
+import StackableContainer from "../../../layout/StackableContainer";
+
+interface EncodeObjectFormProps {
+  readonly setMsgGetter: (msgGetter: MsgGetter) => void;
+  readonly deleteMsg: () => void;
+}
+
+const EncodeObjectForm = ({ setMsgGetter, deleteMsg }: EncodeObjectFormProps) => {
+  const { state } = useAppContext();
+  assert(state.chain.addressPrefix, "addressPrefix missing");
+
+  const [msgTypeUrl, setMsgTypeUrl] = useState("");
+  const [msgValue, setMsgValue] = useState("");
+
+  const [msgTypeUrlError, setMsgTypeUrlError] = useState("");
+  const [msgValueError, setMsgValueError] = useState("");
+
+  useEffect(() => {
+    setMsgTypeUrlError("");
+    setMsgValueError("");
+
+    const isMsgValid = (msg: TxMsg): msg is EncodeObject => {
+      if (!msgTypeUrl) {
+        setMsgTypeUrlError("Type URL is required");
+        return false;
+      }
+
+      if (!msgValue) {
+        setMsgValueError("Value is required");
+        return false;
+      }
+
+      try {
+        JSON.parse(msgValue);
+      } catch {
+        setMsgValueError("Value must be valid JSON");
+        return false;
+      }
+
+      return isEncodeObject(msg);
+    };
+
+    const parsedMsgValue = (() => {
+      try {
+        return JSON.parse(msgValue);
+      } catch {
+        return {};
+      }
+    })();
+
+    const msg: EncodeObject = { typeUrl: msgTypeUrl, value: parsedMsgValue };
```

Comment:
The `value: parsedMsgValue` here is the tricky part: here we need the telescope-specific format. E.g. 64bit integers need to be represented as a `Long` type. Binary data needs to be Uint8Array. There is no standardized way to get this from JSON without message specific JSON -> JavaScript importers.